cfme/storage/manager.py

```diff
@@ -0,0 +1,233 @@
+
+class ManagerEntities(View):
+    """The entities on the main list Storage Manager or Provider page"""
+    table = Table(".//div[@id='list_grid']/table")
+
+
+class ManagerDetailsEntities(View):
+    """The entities on the Storage Manager or Provider details page"""
+    breadcrumb = BreadCrumb()
+    properties = SummaryTable('Properties')
+    relationships = SummaryTable('Relationships')
+    smart_management = SummaryTable('Smart Management')
+    status = SummaryTable('Status')
+
+
+class ManagerDetailsAccordion(View):
+    """The accordion on the Storage Manager or Provider details page"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class ManagerView(BaseLoggedInPage):
+    """A base view for all the Storage Manager or Provider pages"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]')
+
+    @property
+    def in_manager(self):
+        nav_path = self.context['object'].nav.pick(self.context['object'].appliance.version)
+        return(
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == nav_path)
+
+
+class ManagerAllView(ManagerView):
+    """The all Storage Manager or Provider page"""
+    @property
+    def is_displayed(self):
+        return (
+            self.in_manager and
+            self.title.text in ('Storage Managers', self.context['object'].manager_type))
+
+    toolbar = View.nested(ManagerToolbar)
+    entities = View.nested(ManagerEntities)
+    paginator = PaginationPane()
+
+
+class ManagerDetailsView(ManagerView):
+    """The details page for Storage Manager or Provider"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+
+        return(
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(ManagerDetailsToolbar)
+    sidebar = View.nested(ManagerDetailsAccordion)
+    entities = View.nested(ManagerDetailsEntities)
+
+
+class ManagerTagsView(ManagerView):
+    """The tag page for Storage Manager or Provider"""
+    breadcrumb = BreadCrumb()
+    select_tag = BootstrapSelect('tag_cat')
+    select_value = BootstrapSelect('tag_add')
+    save = Button('Save')
+    reset = Button('Reset')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_manager and
+            self.breadcrumb.active_location == 'Tag Assignment')
+
+
+class ManagePoliciesView(ManagerView):
+    """The policies page for Storage Manager or Provider"""
+    breadcrumb = BreadCrumb()
+    policies = BootstrapTreeview("protectbox")
+    save_button = Button("Save")
```

Comment:
REQUIRED: as I already wrote before please either use _button everywhere or remove it everywhere.